### ISSUE TITLE
Namespace conflicts with OSQP constants (was: Osqp library seems to have conflicts with OpenCV library)

### ISSUE THREAD

**Issue by CharlesPeng1998** [osqp/osqp#210]
Both of osqp and opencv are installed by Cmake.
I write a simple code to include these two libraries:

```C++
#include <iostream>
#include <osqp/osqp.h>
#include <opencv2/opencv.hpp>
#include <opencv2/highgui.hpp>

using namespace std;

int main() {
    cout << "Hello" << endl;
    return 0;
}
```

And I write CMakeLists.txt like this:

```cmake
find_package(osqp REQUIRED)
find_package(OpenCV 3.3.0 EXACT)
add_executable(test test.cpp)
target_link_libraries(test PRIVATE osqp::osqp ${OpenCV_LIBS})
```

When I ran make, I got these error messages:
```
In file included from /usr/local/include/osqp/types.h:9:0,
                 from /usr/local/include/osqp/osqp.h:9,
                 from /home/autolab/test/test.cpp:2:
/usr/local/include/opencv2/core/types.hpp:793:9: error: expected identifier before ‘(’ token
         MAX_ITER=COUNT, //!< ditto
         ^
/usr/local/include/opencv2/core/types.hpp:793:9: error: expected ‘}’ before ‘(’ token
/usr/local/include/opencv2/core/types.hpp:793:9: error: expected unqualified-id before numeric constant
         MAX_ITER=COUNT, //!< ditto
         ^
/usr/local/include/opencv2/core/types.hpp:793:9: error: expected ‘)’ before numeric constant
In file included from /usr/local/include/opencv2/core.hpp:58:0,
                 from /usr/local/include/opencv2/opencv.hpp:52,
                 from /home/autolab/test/test.cpp:3:
/usr/local/include/opencv2/core/types.hpp:798:18: error: expected unqualified-id before ‘)’ token
     TermCriteria();
                  ^
/usr/local/include/opencv2/core/types.hpp:804:18: error: expected unqualified-id before ‘int’
     TermCriteria(int type, int maxCount, double epsilon);
                  ^
/usr/local/include/opencv2/core/types.hpp:804:18: error: expected ‘)’ before ‘int’
/usr/local/include/opencv2/core/types.hpp:880:18: error: ‘DataType’ is not a class template
 template<> class DataType<Moments>
                  ^
/usr/local/include/opencv2/core/types.hpp:881:1: error: explicit specialization of non-template ‘DataType’
 {
 ^
/usr/local/include/opencv2/core/types.hpp:888:27: error: ‘DataType’ is not a template
            depth        = DataType<channel_type>::depth,
                           ^
/usr/local/include/opencv2/core/types.hpp:888:27: error: incomplete type ‘DataType’ used in nested name specifier
/usr/local/include/opencv2/core/types.hpp:890:27: error: ‘DataType’ is not a template
            fmt          = DataType<channel_type>::fmt + ((channels - 1) << 8),
                           ^
/usr/local/include/opencv2/core/types.hpp:890:27: error: incomplete type ‘DataType’ used in nested name specifier
/usr/local/include/opencv2/core/types.hpp:894:13: error: ‘Vec’ does not name a type
     typedef Vec<channel_type, channels> vec_type;
             ^
/usr/local/include/opencv2/core/types.hpp:908:1: error: ‘Complex’ does not name a type
 Complex<_Tp>::Complex()
 ^
/usr/local/include/opencv2/core/types.hpp:912:1: error: ‘Complex’ does not name a type
 Complex<_Tp>::Complex( _Tp _re, _Tp _im )
 ^
/usr/local/include/opencv2/core/types.hpp:916:1: error: ‘Complex’ does not name a type
 Complex<_Tp>::operator Complex<T2>() const
.....
```

Hopefully you can provide me with a way to solve this.

**Comment by goulart-paul** [comment]
I would guess that the problem is that the symbol `MAX_ITER` is defined in the OSQP `constants.h` header as `(4000)`, but the same symbol is used within an `enum` in the other library -- specifically in the line `MAX_ITER=COUNT` in your error log.

A simple fix would be to change the `MAX_ITER` symbol everywhere in the OSQP code to something like `OSQP_MAX_ITER`.   

However, even if that works I suggest keeping this issue open for now.   I think it would make sense for OSQP to take more care about defining symbols in our outward facing API that have the potential to create naming conflicts like this.

**Comment by CharlesTuring** [comment]
It works. Thanks.
By the way, the symbol ```RHO``` also cause conflicts with opencv, it can also be fixed in the way you provided.

**Comment by bstellato** [comment]
All names are now prefixed.

**Comment by yifan-hou** [comment]
Where can I get the version with the name fixes? As of today the master branch still has those defines without prefixes: https://github.com/osqp/osqp/blob/master/include/constants.h#L60

**Comment by imciner2** [comment]
The conflicts have been fixed in the `develop-1.0` branch, which is the current development version of the 1.0 release. Note that this contains several breaking API changes (including the namespace conflict fixes), so code using that branch may not natively be compatible with the current 0.6.2 release.
